serde/src/java/org/apache/hadoop/hive/serde2/json/HiveJsonReader.java

```diff
@@ -450,22 +455,36 @@ private Object visitLeafNode(final JsonNode leafNode,
    */
   private byte[] getByteValue(final JsonNode binaryNode) throws SerDeException {
     try {
-      switch (this.binaryEncoding) {
+      BinaryEncoding binaryEncoding = getBinaryEncodingForNode(binaryNode);
+      switch (binaryEncoding) {
       case RAWSTRING:
-        final String byteText = binaryNode.textValue();
+        final String byteText = binaryNode.asText();
+        if (byteText == null) {
+          return null;
```

Comment:
Returning an empty array instead of null can compromise data consistency and risk misinterpreting the data in ways the user may not intend. I don't think we should do so.